src/core/io/processinterfacefb.cpp

```diff
@@ -0,0 +1,227 @@
+/*******************************************************************************
+ * Copyright (c) 2016, 2025 Johannes Messmer (admin@jomess.com), fortiss GmbH,
+ *                          Johannes Kepler University Linz, Thomas Öllinger
+ *                          HR Agrartechnik GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Johannes Messmer - initial API and implementation and/or initial documentation
+ *   Jose Cabral      - cleaning of namespaces
+ *   Alois Zoitl      - further clean-ups and performance improvements on IOHandle
+ *   Thomas Öllinger  - use PARAMS to reference to I/O configuration
+ *   Alois ZOitl      - removed old process interface structure, added common
+ *                      interface elements
+ *******************************************************************************/
+
+#include <processinterfacefb.h>
+#include "criticalregion.h"
+
+using namespace forte::core::io;
+using namespace std::string_literals;
+
+const CIEC_STRING CProcessInterfaceFB::scmOK("OK"_STRING);
+const CIEC_STRING CProcessInterfaceFB::scmWaitingForHandle("Waiting for handle.."_STRING);
+const CIEC_STRING CProcessInterfaceFB::scmFailedToRegister("Failed to register observer."_STRING);
+const CIEC_STRING CProcessInterfaceFB::scmMappedWrongDirectionOutput("Mapped invalid direction. A Q block requires an output handle."_STRING);
+const CIEC_STRING CProcessInterfaceFB::scmMappedWrongDirectionInput("Mapped invalid direction. An I block requires an input handle."_STRING);
+const CIEC_STRING CProcessInterfaceFB::scmMappedWrongDataType("Mapped invalid data type."_STRING);
+
+
+CProcessInterfaceFB::CProcessInterfaceFB(forte::core::CFBContainer &paContainer, const SFBInterfaceSpec& paInterfaceSpec, const CStringDictionary::TStringId paInstanceNameId) :
+    CEventSourceFB(paContainer, paInterfaceSpec, paInstanceNameId),
+    IOObserver(),
+    var_QI(0_BOOL),
```

Comment:
you could have some of these initializations in the header where they are declared